redaction/draft-ietf-trans-redaction.md

```diff
@@ -0,0 +1,229 @@
+---
+title: "Certificate Transparency: Domain Label Redaction"
+docname: draft-ietf-trans-redaction-00
+abbrev: CT Domain Label Redaction
+category: exp
+
+ipr: trust200902
+area: Security
+wg: TRANS (Public Notary Transparency)
+kw: Internet-Draft
+
+stand_alone: yes
+pi: [toc, sortrefs, symrefs]
+
+author:
+ -
+    ins: R. Stradling
+    name: Rob Stradling
+    org: Comodo CA, Ltd.
+    email: rob.stradling@comodo.com
+ -
+    ins: E. Messeri
+    name: Eran Messeri
+    org: Google UK Ltd.
+    email: eranm@google.com
+
+normative:
+  RFC2119:
+  RFC4648:
+  RFC5280:
+  RFC6125:
+  I-D.ietf-trans-rfc6962-bis:
+
+informative:
+  Public.Suffix.List:
+    target: https://publicsuffix.org
+    title: Public Suffix List
+    author:
+      org: Mozilla Foundation
+    date: 2016
+
+--- abstract
+
+We define a mechanism to allow DNS domain name labels that are considered to be
+private to not appear in public Certificate Transparency (CT) logs, while still
+retaining most of the security benefits that accrue from using Certificate
+Transparency mechanisms.
+
+--- middle
+
+# Introduction
+
+Some domain owners regard certain DNS domain name labels within their registered
+domain space as private and security sensitive. Even though these domains are
+often only accessible within the domain owner's private network, it's common for
+them to be secured using publicly trusted Transport Layer Security (TLS) server
+certificates.
+
+Certificate Transparency [I-D.ietf-trans-rfc6962-bis] describes a protocol for
+publicly logging the existence of TLS server certificates as they are issued or
+observed. Since each TLS server certificate lists the domain names that it is
+intended to secure, private domain name labels within registered domain space
+could end up appearing in CT logs, especially as TLS clients develop policies
+that mandate CT compliance. This seems like an unfortunate and potentially
+unnecessary privacy leak, because it's the registered domain names in each
+certificate that are of primary interest when using CT to look for suspect
+certificates.
+
+Section TBD of [I-D.ietf-trans-rfc6962-bis] proposes two mechanisms for dealing
+with this conundrum: wildcard certificates and name-constrained intermediate
+CAs. However, these mechanisms are insufficient to cover all use cases.
```

Comment:
Add a TODO for me to expand on when each of the other mechanisms is suitable and when this mechanism may be suitable.